### PR TITLE
Adding skipIfLocked option to lock step

### PR DESCRIPTION
**Pull request by tomasbjerre**
I previously proposed a `tryLock`-step in #179. This fulfills that requirement with the `lock`-step by adding the option to not wait queue but instead skip the execution of the body.

```groovy
lock(resource: 'some_resource', skipIfLocked: true) {
  echo 'Do something now or never!'
}
```

Fixes part of #107

### CODE REVIEW

**Comment by tomasbjerre** [comment]
Perhaps `skipIfQueue` is better naming.

**Comment by TobiX** [comment]
Nice idea! I'm just a bit unhappy with the name. I always saw the existence of some kind of "queue" more as an artifact/implementation detail. May `skipIfLocked` or something like that? (`skipIfWouldWait` sounds a bit verbose...)

**Comment by tomasbjerre** [comment]
I renamed it to `skipIfLocked`.

File: src/main/resources/org/jenkins/plugins/lockableresources/LockStep/help-skipQueue.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div>
```
Comment (TobiX) [inline src/main/resources/org/jenkins/plugins/lockableresources/LockStep/help-skipQueue.html:1]: Filename does not match field name.

File: src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java [hunk 2]
```diff
@@ -741,7 +745,8 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
   public synchronized Set<LockableResource> checkResourcesAvailability(
       List<LockableResourcesStruct> requiredResourcesList,
       @Nullable PrintStream logger,
-      @Nullable List<String> lockedResourcesAboutToBeUnlocked) {
+      @Nullable List<String> lockedResourcesAboutToBeUnlocked,
+      boolean skipIfLocked) {
```
Comment (TobiX) [inline src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java:749]: Please provide a compatibility function with the old signature - You can then keep the code paths were the new functionality isn't used the same as before.

File: src/main/resources/org/jenkins/plugins/lockableresources/LockStep/help-skipQueue.html (hunk 1 above)
Comment (tomasbjerre) [reply src/main/resources/org/jenkins/plugins/lockableresources/LockStep/help-skipQueue.html]: Renamed to help-skipIfLocked.html

File: src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java (hunk 2 above)
Comment (tomasbjerre) [reply src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java:755]: Done.

**Pull request merged**